src/formatDate.js

```diff
@@ -7,8 +7,61 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newSeparator = toFormat[toFormat.length - 1];
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+
+  const dateParts = date.split(oldSeparator);
+
+  const yearIndexFrom =
+    fromFormat.indexOf('YYYY') !== -1
+      ? fromFormat.indexOf('YYYY')
+      : fromFormat.indexOf('YY');
+  const yearIndexTo =
+    toFormat.indexOf('YYYY') !== -1
+      ? toFormat.indexOf('YYYY')
+      : toFormat.indexOf('YY');
+  const monthIndexFrom = fromFormat.indexOf('MM');
+  const dayIndexFrom = fromFormat.indexOf('DD');
+
+  let year = dateParts[yearIndexFrom];
+  const month = dateParts[monthIndexFrom];
+  const day = dateParts[dayIndexFrom];
+
+  if (fromFormat[yearIndexFrom] === 'YY' && toFormat[yearIndexTo] === 'YYYY') {
+    if (year < 30) {
+      year = `20${year}`;
+    } else {
+      year = `19${year}`;
+    }
+  }
+
+  if (fromFormat[yearIndexFrom] === 'YYYY' && toFormat[yearIndexTo] === 'YY') {
+    year = year.slice(-2);
+  }
+
+  let newDate = '';
+
+  for (const part of toFormat.slice(0, -1)) {
+    switch (part) {
+      case 'YYYY':
+        newDate += year;
+        break;
+      case 'YY':
+        newDate += year;
+        break;
+      case 'MM':
+        newDate += month;
+        break;
+      case 'DD':
+        newDate += day;
+        break;
+    }
+    newDate += newSeparator;
+  }
+
+  return newDate.slice(0, -1);
```

Comment:
The method of removing the trailing separator by slicing the `newDate` string is not the most robust solution. If the `toFormat` array is empty or not provided, this could result in an incorrect output or an error. It would be safer to handle the concatenation of the separator within the loop more carefully to avoid the need for slicing the string at the end.